src/openmrs-backend-dependencies.ts

```diff
@@ -0,0 +1,91 @@
+import { openmrsFetch } from "@openmrs/esm-api";
+import * as semver from "semver";
+import { difference, isEmpty } from "lodash-es";
+
+var installedBackendModules: any[] = [];
+
+function fetchInstalledBackendModules() {
+  return openmrsFetch(`/ws/rest/v1/module?v=custom:(uuid,version)`, {
+    method: "GET"
+  });
+}
+
+function isVersionInstalled(requiredVersion, installedVersion) {
+  return semver.eq(
+    semver.coerce(requiredVersion),
+    semver.coerce(installedVersion)
+  );
+}
+
+function getMissingBackendModules(requiredBackendModules) {
+  const requiredBackendModulesUuids = Object.keys(requiredBackendModules);
+  const installedBackendModuleUuids = installedBackendModules.map(
+    res => res.uuid
+  );
+  return difference(requiredBackendModulesUuids, installedBackendModuleUuids);
+}
+
+function getMismatchedVersions(requiredBackendModules) {
+  const mismatchedModuleVersions: MismatchedModuleVersion[] = [];
+  for (let uuid in requiredBackendModules) {
+    const requiredVersion = requiredBackendModules[uuid];
+    const installedVersion = installedBackendModules.find(
+      module => module.uuid === uuid
+    )["version"];
+    if (!isVersionInstalled(requiredVersion, installedVersion)) {
+      mismatchedModuleVersions.push({
+        uuid,
+        requiredVersion,
+        installedVersion
+      });
+    }
+  }
+  return mismatchedModuleVersions;
+}
+
+function checkIfModulesAreInstalled(requiredBackendModules) {
+  const missingOpenmrsBackendModules = getMissingBackendModules(
+    requiredBackendModules
+  );
+  const modulesWithWrongVersionInstalled = getMismatchedVersions(
+    requiredBackendModules
+  );
+  console.error(
```

Comment:
We will replace the console error messages by either throwing the error or displaying the errors on devtools ui